sdk/src/client/stronghold/secret.rs

```diff
@@ -296,14 +321,19 @@ impl SecretManagerConfig for StrongholdAdapter {
 /// Private methods for the secret manager implementation.
 impl StrongholdAdapter {
     /// Execute [Procedure::BIP39Recover] in Stronghold to put a mnemonic into the Stronghold vault.
-    async fn bip39_recover(&self, mnemonic: String, passphrase: Option<String>, output: Location) -> Result<(), Error> {
+    async fn bip39_recover(
+        &self,
+        mnemonic: Mnemonic,
+        passphrase: Option<String>,
```

Comment:
should this take a `Passphrase` already instead of `String`?

Reply:
I feel like this is the purpose of your PR

Reply:
well, okay, I can change it, too.